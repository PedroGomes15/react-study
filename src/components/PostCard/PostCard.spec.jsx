import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const mock = {
  title: 'test',
  body: 'body test',
  id: 1,
  cover: 'img/img.png',
};

describe('<PostCard/>', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard post={mock} />);

    expect(screen.getByRole('img', { name: 'test' })).toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: 'test' })).toBeInTheDocument();

    expect(screen.getByText('body test')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard post={mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
