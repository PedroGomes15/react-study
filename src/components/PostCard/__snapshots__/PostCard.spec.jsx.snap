// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="test"
    class="post-cover"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      test
    </h2>
    <p>
      body test
    </p>
  </div>
</div>
`;
